[Xcode] Use CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE to read headers vfs file

emw-apple opened the pull request:
#### 86e114abc3e7552c563017b28dfa1a9d605c4645
<pre>
[Xcode] Use CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE to read headers vfs file
<a href="https://rdar.apple.com/114352538">rdar://114352538</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=271009">https://bugs.webkit.org/show_bug.cgi?id=271009</a>

Reviewed by Alex Christensen.

Bind the location of bmalloc and WTF&apos;s VFS file to a build setting, and
use that setting to power generate-tapi-filelist.py in the respective
projects.

The VFS path is a listing of the public and private headers of a target.
When tapi support landed last year, this path was hard-coded. Later
workarounds (272426@main, 272986@main) added some conditional logic to
find the VFS file in various locations but resulted in the input edge on
the &quot;Generate TAPI filelist&quot; script phases being removed. This meant
that the WTF/bmalloc file lists could become stale when a header was
removed from their respective projects.

* Source/WTF/Configurations/WTF.xcconfig:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/bmalloc/Configurations/bmalloc.xcconfig:
* Source/bmalloc/bmalloc.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/276134@main">https://commits.webkit.org/276134@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1fa099ddd04cbb55429da0275f08bcc3d4ce0336

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/43860 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/22907 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/46287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/46500 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/39942 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/26789 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/20307 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/36189 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/44434 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/19940 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/37774 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/17190 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/17462 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/38855 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/1911 "Built successfully") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/37258 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/40031 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/39155 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/48061 "Built successfully") | 
| [❌ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/43461 "Failed to checkout and rebase branch from PR 25910") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/18869 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/15448 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/43002 "Passed tests") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/20263 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/41702 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9750 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/20466 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/50488 "Built successfully") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/19887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/10188 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WTF/Configurations/WTF.xcconfig:
```diff
@@ -42,3 +42,7 @@ ICU_HEADERS_FOLDER_PATH_NO = $(BUILT_PRODUCTS_DIR)$(WK_LIBRARY_HEADERS_FOLDER_PA
+
+// Used to generate headers filelist for TAPI in JavaScriptCore. CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE
+// is not exported to script phases, so it must be bound to a different name.
+WK_HEADERS_VFS_FILE = $(CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE);
```
Why don't we just use the CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE environment variable directly?

emw-apple replied:
I observed that it wasn't being exported to script phases. Didn't dive deep enough to figure out exactly why, but there are a class of build settings that are not automatically exported, so I assume this is one of them.

emw-apple committed:
[Xcode] Use CPP_HEADERMAP_PRODUCT_HEADERS_VFS_FILE to read headers vfs file
rdar://114352538
https://bugs.webkit.org/show_bug.cgi?id=271009

Reviewed by Alex Christensen.

Bind the location of bmalloc and WTF's VFS file to a build setting, and
use that setting to power generate-tapi-filelist.py in the respective
projects.

The VFS path is a listing of the public and private headers of a target.
When tapi support landed last year, this path was hard-coded. Later
workarounds (272426@main, 272986@main) added some conditional logic to
find the VFS file in various locations but resulted in the input edge on
the "Generate TAPI filelist" script phases being removed. This meant
that the WTF/bmalloc file lists could become stale when a header was
removed from their respective projects.

* Source/WTF/Configurations/WTF.xcconfig:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/bmalloc/Configurations/bmalloc.xcconfig:
* Source/bmalloc/bmalloc.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/276134@main
